src/bionics.cpp

```diff
@@ -910,7 +910,7 @@ void player::process_bionic(int b)
 
     // Bionic effects on every turn they are active go here.
     if (bio.id == "bio_night") {
-        if (calendar::turn % 5) {
+        if (calendar::is_time_for(5)) {
```

Comment:
IIRC the artificial night generator uses this to restore the light after the cycle--ratchets up and then back down.  So yeah, this'd need specifically tested.